Invoke add_qt_android_apk during install phase

Earthwings opened the pull request:
I'd like to create an .apk in a single run, and have stripped libraries (no symbols) in the .apk. The command for that is "cmake ...; make install/strip". I specify the stripped libraries as dependencies in add_qt_android_apk. The stripping is done by cmake during install (after the ALL target), so add_qt_android_apk cannot run before it. This patch moves add_qt_android_apk from the ALL target to the install target.

Earthwings committed:
Invoke add_qt_android_apk during install phase

In order to depend on stripped libraries created by the install/strip
cmake target, add_qt_android_apk must run after the corresponding
install targets of the libraries. This is only possible when the
custom target is not part of the default build target (ALL), but
the install target.

LaurentGomila commented:
So when you `make`, nothing is actually created? So you always have to invoke `make install` to have your APK?

Earthwings commented:
Yes, the apk is only created in make install then.

LaurentGomila commented:
This is rather unintuitive. I'm not sure that doing this kind of stuff by default is a good thing.

Earthwings commented:
What if I'd introduce an option (say TARGET_INSTALL) for it?

Earthwings committed:
Introduce a TARGET_INSTALL option to enable apk creation during install

Pass TARGET_INSTALL as argument to add_qt_android_apk to have it create
a target invoked in the install phase. Otherwise the target is run
during the default build phase.

LaurentGomila commented:
I don't know, maybe this is the kind of modification that should be done by the end user rather than as an option of the tool.

Earthwings commented:
Ok, so I modified my CMakeLists.txt to set the EXCLUDE_FROM_ALL property on the target and use the install(CODE ...) command to postpone the build to the install phase. Small hack as cmake docs say setting the EXCLUDE_FROM_ALL property results in undefined behavior when building that target, but it seems to work fine.

Maybe all this should really be part of CPack though, with qt-android-cmake a CPack generator. Seems the more appropriate place.
